exporter/loadbalancingexporter/factory.go

```diff
@@ -32,20 +40,98 @@ func createDefaultConfig() component.Config {
 	otlpDefaultCfg.Endpoint = "placeholder:4317"
 
 	return &Config{
+		TimeoutSettings: exporterhelper.NewDefaultTimeoutConfig(),
+		QueueSettings:   exporterhelper.NewDefaultQueueConfig(),
+		BackOffConfig:   configretry.NewDefaultBackOffConfig(),
 		Protocol: Protocol{
 			OTLP: *otlpDefaultCfg,
 		},
 	}
 }
 
-func createTracesExporter(_ context.Context, params exporter.Settings, cfg component.Config) (exporter.Traces, error) {
-	return newTracesExporter(params, cfg)
+func buildExporterConfig(cfg *Config, endpoint string) otlpexporter.Config {
+	oCfg := cfg.Protocol.OTLP
+	oCfg.Endpoint = endpoint
+	// If top level queue is enabled - per-endpoint queue must be disable
+	// This helps us to avoid unexpected issues with mixing 2 level of exporter queues
+	if cfg.QueueSettings.Enabled {
+		oCfg.QueueConfig.Enabled = false
```

Comment:
Looking at the `ConsumeTraces()`, and it would be similar for metrics and logs. This is the code for when it exports.  

```golang
for exp, td := range exporterSegregatedTraces {
  start := time.Now()
  err := exp.ConsumeTraces(ctx, td)
  exp.consumeWG.Done()
  errs = multierr.Append(errs, err)
  ...
}
```

Now that the `exp` exporter has no queue, won't `ConsumeTraces()` block until exporting is finished? This would, I assume, make the export times much slower because it has to export the routed data in-series. This would take quite a long time especially if there are hundreds of backends to export to, likely causing a timeout.

What might be necessary is an `errgroup` or similar which can execute these now-blocking-calls to the exporter. I would be interested to hear your thoughts or maybe @jpkrohling's or @MovieStoreGuy's thoughts on this.

Reply:
Switching to an export queue also means that includes the queue workers which allow parallel processing, which allows a layer of latency hiding. 

Including a new "queue" worker in this could be hazardous in how it is done, say if you create a routine for each split batch you could end up with high amount of routine scheduling which can cause a significant performance hit. Then if you do more an async send on channels you have the original risk that this solves due to the queue moving from the current queue structure, to now a buffered channel that now needs to be shared again. 

For this iteration, I would suggest leaving it out of this change and keep as an issue in the backlog of "investigate and address if required"

Reply:
Missed that part (
Than yes, disabling seb-exporter queue in code might be too risky for all users
I have removed those lines

Reply:
@Fiery-Fenix @MovieStoreGuy revisiting this, I think we actually do need to have a way of disabling the sub exporter queue without blocking in series here.

It's not really solving the problem if the sub exporter has a queue remaining, because that's where the data loss occurred. Then, if we disable the sub-exporter queue, it's extremely slow (blocking exporting in series for each endpoint). 

So I think this part of the code needs more work. Perhaps we should re open the issue or start a new issue.